### PR TITLE
bring support for subkeys (using .)

### PR DESCRIPTION
**Pull request by j0code**
Adds support for dotted config keys.

Example: `smtp.port`
In config.json, this will correspond to:
```json
{
  "smtp": {
    "port": 21
  }
}
```
In the environment, both `smtp.port` and `smtp_port` will be accepted.

### CODE REVIEW

**Review by cfpwastaken — APPROVED** [review]

**Review by cfpwastaken — APPROVED** [review]
Ah thats very cool

**Review by cfpwastaken — APPROVED** [review]

**Review by cfpwastaken — APPROVED** [review]

File: src/config.ts [hunk 1]
```diff
@@ -31,6 +31,23 @@ export default function getConfig<T extends ConfigValue>(key: string, defaultVal
...
+	const value = process.env[key] ?? process.env[key.replaceAll(".", "_")]
+	if (!value) return
+
+	return parseStringValue(value)
+}
+
+function parseStringValue(value: string): ConfigValue {
+	// boolean
+	if (value == "true") return true
+	else if (value == "false") return false
```
Comment (cfpwastaken) [inline src/config.ts:44]: if(Boolean(value)) return !!value;
would something like that work?

Comment (j0code) [reply src/config.ts:44]: no, any string that isn't empty ("") is a truthy value
so Boolean(value) would always be true and it'll return true (Boolean(x) does the same as !!x)
I can
```suggestion
	if (["true", "false"].includes(value)) return !!value
```
tho
but it doesn't really become any more readable that way 🤷‍♀️

Comment (cfpwastaken) [reply src/config.ts:44]: ok let's keep it

**Review by cfpwastaken — APPROVED** [review]

**Pull request merged**